src/cljc/matcher_combinators/core.cljc

```diff
@@ -143,6 +143,16 @@
       issue
       (compare-maps expected actual model/->Unexpected false))))
 
+(defrecord EqualsRecord [expected]
+  Matcher
+  (match [_this actual]
+    (if-let [issue (validate-input expected actual map? 'equals "record")]
+      issue
+      (cond
+        (and (record? expected)
+             (record? actual)) (match (->Value expected) actual)
```

Comment:
Mismatches for the same record type with different values doesn't show nice diffs. 
```
        [:mismatch (mismatch {:x 1, :y 2} {:x 2, :y 2})]
```
could be
```
        [:mismatch {:x (mismatch 1 2), :y 2}]
```
for 
```
(fact (->Point 2 2) => (match (m/equals (->Point 1 2))))
```

Reply:
Also, I wonder if it is better to make is such that if the `expected` is a record then the match fails if the `actual` isn't.

Reply:
So maybe something along the lines of 

```clojure
(defn- match-record [expected actual]
  (if (and (record? actual)
           (= (type expected) (type actual)))
    (match (->EqualsMap expected) actual)
    {::result/type   :mismatch
     ::result/value  (model/->RecordMismatch expected actual)
     ::result/weight 1}))

(defrecord EqualsRecord [expected]
  Matcher
  (match [_this actual]
    (if-let [issue (validate-input expected actual map? 'equals "record")]
      issue
      (cond
        (record? expected)
        (match-record expected actual)

        :else
        (match (->EqualsMap expected) actual)))))
```

Reply:
☝️ this would be a breaking change, but makes sense.